### PR TITLE
Adding a vignette, revised readme. and pkgdown documentation

### PR DESCRIPTION
**Pull request by dshkol**
Vignette covers some basic usage, mostly copied from the readme and made more generic. 

To deploy pkgdown on github pages, can do it in a [few different ways](https://sahirbhatnagar.com/blog/2020/03/03/creating-a-website-for-your-r-package/)

The easiest way is simply to push the docs folder with all the material. I took it out of the .gitignore. Then in the settings for the repo on github just have it deploy from the master/docs folder. (One small but topical issue, this seems to break for now if you rename the master repo to something else). The harder way is to add an action to travis-CI to trigger the deploy. I've generally done the former on my packages.

### CODE REVIEW

File: vignettes/geohashTools.Rmd [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  geom_sf(data = meuse_sf) +
+  geom_sf(data = meuse_gh, fill  = NA) +
+  geom_sf_text(data = meuse_gh, aes(label = rownames(meuse_gh))) +
+  theme_void()
+```
+
+Alternatively, using `gh_covering` with the parameter `minimal = TRUE` will only create geohashes for intersecting objects. 
+```{r meuse_minimal}
+# Currently doesn't work
+# meuse_gh <- gh_covering(meuse_sf, minimal = TRUE)
```
Comment (MichaelChirico) [inline vignettes/geohashTools.Rmd]: this works now right?

**Comment by MichaelChirico** [comment]
Travis failed:

> there is no package called 'ggplot2'

We need to upgrade the `r_packages` in `.travis.yml`

File: DESCRIPTION [hunk 2]
```diff
@@ -9,4 +9,11 @@ Depends: R (>= 3.0.0)
...
 URL: https://github.com/MichaelChirico/geohashTools
 License: MPL-2.0 | file LICENSE
-Suggests: rgdal, sf, sp, testthat
+Suggests: 
+    rgdal,
+    sf,
+    sp,
+    testthat,
+    knitr,
+    rmarkdown
```
Comment (MichaelChirico) [inline DESCRIPTION]: From [WRE](https://cran.r-project.org/doc/manuals/r-release/R-exts.html), we'll need `ggplot2` here as well if it's used in the vignette:

> The ‘Suggests’ field uses the same syntax as ‘Depends’ and lists packages that are not necessarily needed. This includes packages used only in examples, tests or vignettes (see Writing package vignettes), and packages loaded in the body of functions. E.g., suppose an example11 from package foo uses a dataset from package bar. Then it is not necessary to have bar use foo unless one wants to execute all the examples/tests/vignettes: it is useful to have bar, but not necessary. Version requirements can be specified but should be checked by the code which uses the package.

**Comment by dshkol** [comment]
I’ll rework without ggplot to reduce dependence, although I assume every R
user has ggplot2 installed.

File: vignettes/geohashTools.Rmd [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+plot(merlion_nbhd)
+```
+
+### gh_covering
+
+Sometimes we have a set of spatial datapoints that we want to aggregate or index using geohashes. The `gh_covering` function produces a grid of geohashes that overlap with the spatial points. For this function to work, the spatial object has to be in the [WGS84 (EPSG 4326)](https://epsg.io/4326) coordinate reference system that the geohashes use. 
+
+Let's use the included `meuse` dataset in the `sp` package which shows point coordinates for metal deposits. The first element in this object shows values for cadmium deposits. 
+```{r meuse, fig.width = 6, fig.height = 6, out.width = '\\textwidth'}
+library(ggplot2)
```
Comment (MichaelChirico) [inline vignettes/geohashTools.Rmd:127]: I think we should treat this more as a suggested package, and branch out like

```
if (!requireNamespace(ggplot2)) {
  # either attempt to install it, or stop()
}

File: _pkgdown.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+home:
+  title: Tools for working with geohashes
+authors:
+  Michael Chirico:
+    href: https://github.com/MichaelChirico
+template:
+  params:
```
Comment (MichaelChirico) [inline _pkgdown.yml:7]: for the `pkgdown` site, I would deploy it on GitHub pages:

https://pkgdown.r-lib.org/reference/deploy_site_github.html

Comment (dshkol) [reply _pkgdown.yml:7]: Fixed in a3c4fde

File: DESCRIPTION (hunk 2 above)
Comment (dshkol) [reply DESCRIPTION]: Fixed in a3c4fde

File: vignettes/geohashTools.Rmd (hunk 1 above)
Comment (dshkol) [reply vignettes/geohashTools.Rmd]: Yep, confirmed in 26db6f5

File: vignettes/geohashTools.Rmd (hunk 3 above)
Comment (dshkol) [reply vignettes/geohashTools.Rmd:127]: > Travis failed:
> 
> > there is no package called 'ggplot2'
> 
> We need to upgrade the `r_packages` in `.travis.yml`

Fixed in a3c4fde

Comment (dshkol) [reply vignettes/geohashTools.Rmd:127]: > I think we should treat this more as a suggested package, and branch out like
> 
> ```
> if (!requireNamespace(ggplot2)) {
>   # either attempt to install it, or stop()
> }
> ```

Added in 26db6f5

**Comment by MichaelChirico** [comment]
LGTM. merging. let's see if anything needs to change for the site to deploy on GH pages

**Pull request merged**